envs/quadrotors/quad_multi_model.py

```diff
@@ -13,25 +14,30 @@ def __init__(self, cfg, obs_space, timing, self_obs_dim=18, neighbor_obs_dim=6,
         self.self_obs_dim = self_obs_dim
         self.neighbor_obs_dim = neighbor_obs_dim
         self.neighbor_hidden_size = neighbor_hidden_size
+        self.use_spectral_norm = cfg.use_spectral_norm
 
         fc_encoder_layer = cfg.hidden_size
         # encode the current drone's observations
         self.self_encoder = nn.Sequential(
-            nn.Linear(self.self_obs_dim, fc_encoder_layer),
+            fc_layer(self.self_obs_dim, fc_encoder_layer, spec_norm=self.use_spectral_norm),
             nonlinearity(cfg),
-            nn.Linear(fc_encoder_layer, fc_encoder_layer),
+            fc_layer(fc_encoder_layer, fc_encoder_layer, spec_norm=self.use_spectral_norm),
             nonlinearity(cfg)
         )
+
         # encode the neighboring drone's observations
         self.neighbor_encoder = nn.Sequential(
-            nn.Linear(self.neighbor_obs_dim, self.neighbor_hidden_size),
+            fc_layer(self.neighbor_obs_dim, self.neighbor_hidden_size, spec_norm=self.use_spectral_norm),
             nonlinearity(cfg),
+            fc_layer(self.neighbor_hidden_size, self.neighbor_hidden_size, spec_norm=self.use_spectral_norm),
+            nonlinearity(cfg)
         )
+
         self.self_encoder_out_size = calc_num_elements(self.self_encoder, (self.self_obs_dim,))
         self.neighbor_encoder_out_size = calc_num_elements(self.neighbor_encoder, (self.neighbor_obs_dim,))
 
         # Feed forward self obs and neighbor obs after concatenation
-        self.feed_forward = nn.Linear(self.self_encoder_out_size + self.neighbor_encoder_out_size, cfg.hidden_size)
+        self.feed_forward = fc_layer(self.encoder_out_size + self.neighbor_encoder_out_size, cfg.hidden_size, spec_norm=self.use_spectral_norm)
```

Comment:
We can not change `self.self_encoder_out_size` to `self.encoder_out_size`, right? Since we have never defined `self.encoder_out_size`. I would fix it and create a PR.